### DISCUSSION TITLE
I think that the voices are not capable of reading or intoning interrogative sentences correctly.

### DISCUSSION THREAD

**Discussion by laramaceo** [rhasspy/piper#477] [General]
Hello,

The quality of Piper tts has caught my attention. The voices sound realistic, but I think there is an aspect to improve in terms of the voices. I think that the voices are not capable of reading or intoning interrogative sentences correctly. I've made comparisons between Piper's voices and

  https://www.narakeet.com/app/text-to-audio/?projectId=169d71ef-3999-4f5f-8329-8463ef5762d7

The voices on this website are capable of reading and intoning interrogative sentences correctly. My language is Spanish and at least to me the difference in this aspect is evident.

I invite you to make your own comparison in your language.

  If Piper tts improved this deficiency for all languages, it would be even more spectacular!

And exclamatory sentences? (-----((( 👈 😁👍

**Comment by rmcpantoja** [comment]
Hi @laramaceo, 
Intonations in question marks and exclamations largely depend on whether the dataset used to train that voice reflects those features, both in the audio and in the speaker's transcription. My language is Spanish too, I trained better datasets reflecting those features and works great. I wrote [guidelines](https://github.com/rmcpantoja/tTS-dataset-guidelines) for this purpose. Anyway, I'm planning to experiment with adding pitch predictions to the model architecture.

**Reply by laramaceo** [reply]
¡Excelente!, gracias por la respuesta. Me alegra que tu idioma sea también el español, y que estés intentando agregar esta característica, esto me da esperanza de tener un mejor motor tts y que no tenga nada que envidiar a la competencia de pago. Por cierto, en estos momentos solo estoy usando una chromebook con Google tts este, carece también de esta característica. En un futuro cercano planeo tener una nueva PC e instalar Linux y Piper tts.
Por ahora solo tengo como referencia las voces de:

https://huggingface.co/spaces/HirCoir/Piper-TTS-Spanish

Me gustaría luego probar ese nuevo cambio cuando esté disponible. Gracias una vez más y éxito en tu proyecto.
